R/shiny.R

```diff
@@ -31,3 +30,29 @@ shiny_input <- function(input_id, shiny_ui, value = NULL) {
     shiny_ui
   )
 }
+
+#' Universal binding for Shiny input on custom user interface with text value. Main difference between this and shiny_input is lack of
+#' JSON serialization.
+#'
+#' @param input_id String with name of this input. Access to this input within server code is normal with input[[input_id]].
+#' @param shiny_ui UI of HTML component presenting this input to the users. This UI should allow to extract its value with jQuery $().val() function.
+#' @param value An optional argument with value that should be set for this input. Can be used to store persisten input valus in dynamic UIs.
+#'
+#' @export
+shiny_text_input <- function(input_id, shiny_ui, value = NULL) {
+  selected <- shiny::restoreInput(id = input_id, default = value)
+
+  custom_input_class <- 'shiny-custom-text-input'
+  shiny_ui$attribs$class <- paste(custom_input_class, ifelse(is.null(shiny_ui$attribs$class), "", shiny_ui$attribs$class))
```

Comment:
this is a lot of duplicated logic -- it'll bite us very hard when we keep building this package; please extract the common part of logic from these two functions